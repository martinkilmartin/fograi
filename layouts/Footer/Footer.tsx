import { Copyright } from '@components/Copyright';
import { FOOTER_LINKS } from '@constants/FOOTER_LINKS';
import Link from 'next/link';

const Footer = (): JSX.Element => (
  <footer>
    <div
      className="grid-container"
      style={{ alignContent: 'center', padding: '2px' }}
    >
      {FOOTER_LINKS.map((linkSection, idx) => (
        <div className="grid" key={idx}>
          <div
            style={{
              display: 'flex',
              flexDirection: 'column',
            }}
          >
            <h3>{linkSection.title}</h3>
            {linkSection.urls.map((url, idx) => (
              <span style={{ fontSize: 18, fontWeight: 'bold' }} key={idx}>
                <Link href={url.path}>{url.text}</Link>
              </span>
            ))}
          </div>
        </div>
      ))}
    </div>
    <div>
      <Copyright />
    </div>
    <style jsx>{`
      footer {
        width: 100%;
        bottom: 0;
        padding-bottom: 10px;
        text-align: center;
      }
    `}</style>
  </footer>
);

export default Footer;
