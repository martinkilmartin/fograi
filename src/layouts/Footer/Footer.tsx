import { Grid, Text } from '@nextui-org/react';
import { Copyright } from '@components/Copyright';
import { FOOTER_LINKS } from '@constants/FOOTER_LINKS';
import Link from 'next/link';
import { RedditLogo } from '@components/SVG';
import X from '@components/SVG/X';

const Footer = (): JSX.Element => (
  <footer>
    <Grid.Container justify="center" gap={2}>
      {FOOTER_LINKS.map((linkSection, idx) => (
        <Grid key={idx}>
          <div
            style={{
              display: 'flex',
              flexDirection: 'column',
            }}
          >
            <h3>{linkSection.title}</h3>
            {linkSection.urls.map((url, idx) => (
              <Text key={idx} weight="bold" size={18}>
                <Link href={url.path}>{url.text}</Link>
              </Text>
            ))}
          </div>
        </Grid>
      ))}
    </Grid.Container>
    <Grid.Container justify="center" gap={2}>
      <Grid>
        <Text>
          <Link href="https://www.reddit.com/u/noozenews" target="_blank">
            <RedditLogo />
          </Link>
        </Text>
      </Grid>
      <Grid>
        <Text>
          <Link href="https://twitter.com/NoozeNews" target="_blank">
            <X />
          </Link>
        </Text>
      </Grid>
    </Grid.Container>
    <div>
      <Copyright />
    </div>
    <style jsx>{`
      footer {
        width: 100%;
        bottom: 0;
        padding-bottom: 10px;
        text-align: center;
      }
    `}</style>
  </footer>
);

export default Footer;
