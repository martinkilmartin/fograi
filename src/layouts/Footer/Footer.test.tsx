import { BRAND } from '@constants/CONTENT'
import renderer from 'react-test-renderer'
import Footer from './Footer'

test('Footer displays copyright with current year and brand', () => {
  const footerRenderer = renderer.create(<Footer />)
  const tree = footerRenderer.toJSON()
  expect(tree).toMatchSnapshot()
  const footerInstance = footerRenderer.root
  const footerElement = footerInstance.findByProps({
    className: 'p-10 footer bg-primary text-primary-content footer-center',
  })
  expect(footerElement.findByType('p').children[0]).toEqual(
    `Â© ${new Date().getFullYear()} ðŸ‡®ðŸ‡ª ${BRAND} ðŸ‡¨ðŸ‡¦`
  )
})
