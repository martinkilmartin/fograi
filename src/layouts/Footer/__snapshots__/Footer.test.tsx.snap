// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer displays copyright with current year and brand 1`] = `
<footer
  className="p-10 footer bg-primary text-primary-content footer-center"
>
  <div
    className="grid-rows footer"
  >
    <div>
      <span
        className="text-xl opacity-100 footer-title"
      >
        Info
      </span>
      <a
        className="text-lg link link-hover"
        href="/about"
      >
        About
      </a>
      <a
        className="text-lg link link-hover"
        href="/contact"
      >
        Contact
      </a>
      <a
        className="text-lg link link-hover"
        href="/sources"
      >
        Sources
      </a>
    </div>
    <div>
      <span
        className="text-xl opacity-100 footer-title"
      >
        Legal
      </span>
      <a
        className="text-lg link link-hover"
        href="/cookie"
      >
        Cookie Policy
      </a>
      <a
        className="text-lg link link-hover"
        href="/privacy"
      >
        Privacy Policy
      </a>
      <a
        className="text-lg link link-hover"
        href="/terms"
      >
        Terms Of Use
      </a>
    </div>
  </div>
  <div>
    <p>
      © 2021 Fógraí
    </p>
  </div>
</footer>
`;
